pkg/operator/controller/ingress/controller.go

```diff
@@ -352,7 +352,7 @@ func setDefaultDomain(ic *operatorv1.IngressController, ingressConfig *configv1.
 }
 
 func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig *configv1.Infrastructure) bool {
-	effectiveStrategy := ic.Spec.EndpointPublishingStrategy
+	effectiveStrategy := ic.Spec.EndpointPublishingStrategy.DeepCopy()
```

Comment:
Is there any need to backport this semantic change?

Reply:
No, this change only fixes an issue revealed by unit tests.  The change isn't needed to resolve the original issue in the BZ.

Reply:
If we backport the fix, I'd rather backport just #681 and not #691.  The deep copy is to fix a correctness issue revealed by the unit tests but isn't required to fix the original issue.